bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/AbstractSessionEventConsumer.java

```diff
@@ -69,21 +66,22 @@ public AbstractSessionEventConsumer(ISessionEventProducer ssp) {
      * @param sse an <code>ExecEvent</code>
      */
     public void sessionUpdate(ExecEvent sse) {
+        LOG.info(String.format("%s: enqueueing event: %s", getName(), sse));
         // Place on queue
         _evq.add(sse);
     }
 
-    public void stop() {
-        _done = true;
-    }
+    public abstract String getName();
 
     // Runnable
     public void run() {
         // Remove from queue and run
-        while (!_done) {
+        while (true) {
```

Comment:
Is this correct? we run this forever?

Reply:
Apparently.  There were no usages of `stop` and anyway it wasn't implemented correctly.